locations/spiders/pep.py

```diff
@@ -0,0 +1,66 @@
+from scrapy import Spider
+from scrapy.http import JsonRequest
+
+from locations.dict_parser import DictParser
+from locations.hours import OpeningHours
+
+
+class PepSpider(Spider):
+    name = "pep"
+    allowed_domains = ["pepstores.com"]
+    start_urls = ["https://www.pepstores.com/graphql?operationName=nearbyStores"]
+
+    def start_requests(self):
+        graphql_query = """query nearbyStores($brandId: String, $latitude: Float, $longitude: Float, $radius: Int) {
+          nearbyStores(
+            brandId: $brandId
+            latitude: $latitude
+            longitude: $longitude
+            radius: $radius
+          ) {
+              address
+              branch_id
+              business_hours
+              description
+              distance
+              latitude
+              longitude
+              store_brand
+              telephone_number
+              __typename
+            }
+        }"""
+        data = {
+            "operationName": "nearbyStores",
+            "variables": {"brandId": "pep", "latitude": -22, "longitude": 24, "radius": 2000},
+            "query": graphql_query,
+        }
+        for url in self.start_urls:
+            yield JsonRequest(url=url, method="POST", data=data)
+
+    def parse(self, response):
+        for location in response.json()["data"]["nearbyStores"]:
+            item = DictParser.parse(location)
+
+            item["name"] = location.get("description")
+
+            if location.get("description").startswith("PEP Cell"):
+                item["brand"] = "Pep Cell"
+                item["brand_wikidata"] = "Q128802743"
+            elif location.get("description").startswith("PEP Home"):
+                item["brand"] = "Pep Home"
+                item["brand_wikidata"] = "Q128802022"
+            else:
+                item["brand"] = "Pep"
+                item["brand_wikidata"] = "Q7166182"
+
+            item["opening_hours"] = OpeningHours()
+            item["opening_hours"].add_ranges_from_string(
+                # 00H00-00H00 was interpreted as being open all day, but has the opposite meaning
+                location["business_hours"]
+                .replace("00H00-00H00", "closed")
+                .replace("\\n", " ")
+                .replace("H", ":")
```

Comment:
cc @davidhicks

Reply:
Note to self that `add_ranges_from_string` should prevent an opening time and closing time being the same value (0:00 in particular). I'm the process of rewriting `add_ranges_from_string` to be better locale-aware and more customisable, so will try and remember to include this test case.

Reply:
That's a problem as well, but I realise I should have been clearer, because I was actually meaning the fact that the time ranges were being given as e.g. "Mon: 09H00-17H00" which was not being parsed without replacing the "H" with ":". Either this should be checked for, or I could specify the time format, "%HH%M"